tool/tsh/mfa.go

```diff
@@ -157,11 +162,33 @@ func newMFAAddCommand(parent *kingpin.CmdClause) *mfaAddCommand {
 }
 
 func (c *mfaAddCommand) run(cf *CLIConf) error {
+	tc, err := makeClient(cf, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	ctx := cf.Context
+
+	deviceTypes := deviceTypes
 	if c.devType == "" {
-		// TODO(codingllama): Revisit CLI nomenclature, we only talk about U2F on
-		//  login.
-		var err error
-		c.devType, err = prompt.PickOne(cf.Context, os.Stdout, prompt.Stdin(), "Choose device type", deviceTypes)
+		// If we are prompting the user for the device type, then take a glimpse at
+		// server-side settings and adjust the options accordingly.
+		// This is undesirable to do during flag setup, but we can do it here.
+		pingResp, err := tc.Ping(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		log.Debugf("Got server-side preferred local MFA: %v", pingResp.Auth.PreferredLocalMFA)
+		switch preferredMFA := pingResp.Auth.PreferredLocalMFA; preferredMFA {
+		case constants.SecondFactorOTP: // OTP only
+			deviceTypes = []string{totpDeviceType}
+		case constants.SecondFactorU2F: // OTP+U2F
+			deviceTypes = []string{totpDeviceType, u2fDeviceType}
+		case constants.SecondFactorWebauthn: // OTP+WebAuthn
+			deviceTypes = []string{totpDeviceType, webauthnDeviceType}
```

Comment:
hm.. i am not sure i understand why totp device is included in both?

Reply:
oh i see, if sf is `optional` or `on` it returns webauthn or u2f, but how would you know if it's only u2f or only webauthn?

Reply:
Usually tsh doesn't filter suggestions, instead it lets the backend reject unwanted device types. That said, this is looking wonky, let me try to do better.

Reply:
I attempted to deal with the various nuances of second factor configuration and decided, in the end, that is was not worth it. We now use preferredMFA only to pick between U2F and Webauthn and let the backend deal with the rest.

Let me know if it looks better.